From 1017b0ceba880bcf0fc6f12c434083c01cc19235 Mon Sep 17 00:00:00 2001
From: Vitalii Vorobiov <vi.vorobiov@samsung.com>
Date: Thu, 7 Jul 2016 20:31:01 +0300
Subject: [PATCH 77/79] edje_edit: on save_all firstly save all cached group,
 then all other

this will fix case when aliased group was changed and alias-group was loaded,
so then on saving alias-group WON'T rewrite all changed to aliased group

@fix
---
 src/lib/edje/edje_edit.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index c1a2fef..94ee615 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -15914,6 +15914,17 @@ _edje_edit_internal_save(Evas_Object *obj, int current_only, Eina_Bool generate_
 
         INF("** Writing all collections");
 
+        EINA_LIST_FOREACH(ef->collection_cache, l, edc)
+          {
+             INF("** Writing cache Edje_Part_Collection* ed->collection "
+                 "[id: %d]", edc->id);
+             if (!_edje_edit_collection_save(eetf, edc))
+               {
+                  eet_close(eetf);
+                  return EINA_FALSE;
+               }
+          }
+
         it = eina_hash_iterator_data_new(ef->collection);
         while (eina_iterator_next(it, (void **)&ce))
           {
@@ -15929,17 +15940,6 @@ _edje_edit_internal_save(Evas_Object *obj, int current_only, Eina_Bool generate_
                }
           }
         eina_iterator_free(it);
-
-        EINA_LIST_FOREACH(ef->collection_cache, l, edc)
-          {
-             INF("** Writing cache Edje_Part_Collection* ed->collection "
-                 "[id: %d]", edc->id);
-             if (!_edje_edit_collection_save(eetf, edc))
-               {
-                  eet_close(eetf);
-                  return EINA_FALSE;
-               }
-          }
      }
 
    if ((eed->bytecode_dirty || eed->script_need_recompile) && ed->collection)
-- 
2.7.4

